src/pycrunch/importing.py

```diff
@@ -53,13 +53,13 @@ def wait_for_batch_status(self, batch, status):
             raise ValueError("The batch did not reach the '%s' state in the "
                              "given time. Please check again later." % status)
 
-    def add_source(self, ds, filename, fp, mimetype):
+    def add_source(self, ds, filename, fp, mimetype, data={}):
         """Create a new Source on the given dataset and return its URL."""
         sources_url = ds.user_url.catalogs['sources']
         # Don't call Catalog.post here (which would force application/json);
         # we want requests.Session to set multipart/form-data with a boundary.
         new_source_url = ds.session.post(
-            sources_url, files={"uploaded_file": (filename, fp, mimetype)}
+            sources_url, files={"uploaded_file": (filename, fp, mimetype)}, data=data
```

Comment:
What is this data argument? The POST method on the sources controller does not specify such argument.

Reply:
The sources controller now expects data in the POST body which is where we are sending the required attributes 
for the crunchlake source. 

`data` is not a sources controller specific attribute. It's request body data for the POST request.

Reply:
Ah, got it, this is not a `data` query param, but the request data